contracts/src/CW721ERC721Wrapper.sol

```diff
@@ -0,0 +1,131 @@
+        bytes memory response = WasmdPrecompile.query(Cw721Address, bytes(req));
+        bytes memory approval = JsonPrecompile.extractAsBytes(response, "approval");
+        bytes memory res = JsonPrecompile.extractAsBytes(approval, "spender");
+        return AddrPrecompile.getEvmAddr(string(res));
+    }
+
+    function isApprovedForAll(address owner, address operator) public view override returns (bool) {
+        string memory o = _formatPayload("owner", _doubleQuotes(AddrPrecompile.getSeiAddr(owner)));
+        string memory req = _curlyBrace(_formatPayload("all_operators", _curlyBrace(o)));
+        bytes memory response = WasmdPrecompile.query(Cw721Address, bytes(req));
+        bytes[] memory approvals = JsonPrecompile.extractAsBytesList(response, "operators");
+        for (uint i=0; i<approvals.length; i++) {
+            bytes memory op = JsonPrecompile.extractAsBytes(approvals[i], "spender");
+            if (AddrPrecompile.getEvmAddr(string(op)) == operator) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    // Calls
+    function transferFrom(address from, address to, uint256 tokenId) public override {
+        if (to == address(0)) {
+            revert ERC721InvalidReceiver(address(0));
+        }
+        string memory recipient = _formatPayload("recipient", _doubleQuotes(AddrPrecompile.getSeiAddr(to)));
+        string memory tId = _formatPayload("token_id", _doubleQuotes(Strings.toString(tokenId)));
+        string memory req = _curlyBrace(_formatPayload("transfer_nft", _curlyBrace(_join(recipient, tId, ","))));
+        _execute(bytes(req));
+        emit Transfer(from, to, tokenId);
+    }
+
+    function approve(address approved, uint256 tokenId) public override {
+        string memory spender = _formatPayload("spender", _doubleQuotes(AddrPrecompile.getSeiAddr(approved)));
+        string memory tId = _formatPayload("token_id", _doubleQuotes(Strings.toString(tokenId)));
+        string memory req = _curlyBrace(_formatPayload("approve", _curlyBrace(_join(spender, tId, ","))));
+         _execute(bytes(req));
+         emit Approval(ownerOf(tokenId), approved, tokenId);
+    }
+
+    function setApprovalForAll(address operator, bool approved) public override {
+        string memory op = _curlyBrace(_formatPayload("operator", _doubleQuotes(AddrPrecompile.getSeiAddr(operator))));
+        if (approved) {
+            _execute(bytes(_curlyBrace(_formatPayload("approve_all", op))));
+        } else {
+            _execute(bytes(_curlyBrace(_formatPayload("revoke_all", op))));
+        }
+        emit ApprovalForAll(msg.sender, operator, approved);
+    }
+
+    function _execute(bytes memory req) internal returns (bytes memory) {
+        (bool success, bytes memory ret) = WASMD_PRECOMPILE_ADDRESS.delegatecall(
+            abi.encodeWithSignature(
+                "execute(string,bytes,bytes)",
+                Cw721Address,
+                bytes(req),
+                bytes("[]")
```

Comment:
Can you explain why this param is just blank? When would it not be blank?

Reply:
this param is the amount of funds that would be sent to the CW contracts. In the case of ERC721 it should always send no fund